components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1000,7 +1000,7 @@ void ConversationHandler::RespondToToolUseRequest(
 
   tool_use->output = std::move(output);
 
-  OnHistoryUpdate(chat_history_.back()->Clone());
```

Comment:
why do we replace OnHistoryUpdate instead of append OnToolUseEventOutput

Reply:
Efficiency because nothing needs to know about the change to the existing ToolUseEvent apart from something subscribed to `OnToolUseEventOutput`.  And there might be a lot of tool use events with a lot of images. And semantically, the history hasn't changed at this point.
In fact, we shouldn't have the parent frame receive OnHistoryUpdate at all, we should just send the inferred state properties directly since it doesn't display history.